tests/runtimes/android/android_runtime_tests.py

```diff
@@ -0,0 +1,271 @@
+
+    def test_318_generated_classes_not_be_deleted_on_rebuild(self):
+        """
+            https://github.com/NativeScript/nativescript-cli/issues/3560
+        """
+        Tns.platform_remove(app_name=APP_NAME, platform=Platform.ANDROID)
+        Tns.platform_add_android(APP_NAME, framework_path=Android.FRAMEWORK_PATH)
+        target = os.path.join(TEST_RUN_HOME, APP_NAME, 'app')
+        source = os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files', 'android-runtime-904',
+                              'MyActivity.js')
+        File.copy(source, target)
+
+        Tns.build_android(os.path.join(TEST_RUN_HOME, APP_NAME))
+
+        assert File.exists(os.path.join(TEST_RUN_HOME, APP_NAME, "app", "MyActivity.js"))
+        assert File.exists(
+            os.path.join(TEST_RUN_HOME, APP_NAME, "platforms", "android", "app", "src", "main", "java", "com", "tns",
+                         "MyActivity.java"))
+
+        File.delete(os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'MyActivity.js'))
+
+        Tns.build_android(os.path.join(TEST_RUN_HOME, APP_NAME))
+
+        assert not File.exists(
+            os.path.join(TEST_RUN_HOME, APP_NAME, "platforms", "android", "app", "src", "main", "java", "com", "tns",
+                         "MyActivity.java"))
+
+    def test_350_sgb_fails_generating_custom_activity(self):
+        """
+        Static Binding Generator fails if class has static properties that are used within the class
+        https://github.com/NativeScript/android-runtime/issues/1160
+        """
+        Folder.clean(os.path.join(TEST_RUN_HOME, APP_NAME))
+        Tns.create(app_name=APP_NAME, template=Template.HELLO_WORLD_JS.local_package, update=True)
+        Tns.platform_add_android(APP_NAME, framework_path=Android.FRAMEWORK_PATH)
+
+        source = os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files', 'android-runtime-1160',
+                              'testActivity.android.js')
+        target = os.path.join(TEST_RUN_HOME, APP_NAME, 'app')
+        File.copy(src=source, target=target)
+
+        Tns.build_android(os.path.join(TEST_RUN_HOME, APP_NAME))
+        activity_class_path = os.path.join(TEST_RUN_HOME, APP_NAME, "platforms", "android", "app", "src", "main",
+                                           "java", "com",
+                                           "test")
+
+        if File.exists(os.path.join(activity_class_path, "Activity.java")):
+            assert True
+        else:
+            assert False, "Fail: Custom activity class is NOT generated in {0} !".format(activity_class_path)
+
+    def test_390_code_cache_option(self):
+        """
+        CodeCache option is broken since Android Runtime 4.1.0
+
+        https://github.com/NativeScript/android-runtime/issues/1235
+        """
+        Folder.clean(os.path.join(TEST_RUN_HOME, APP_NAME))
+        Tns.create(app_name=APP_NAME, template=Template.HELLO_WORLD_JS.local_package, update=True)
```

Comment:
Can we use Template.MIN_JS?

Reply:
All tests are based on hello world js template. I would prefer using it.

Reply:
You will wait ~20sec more only on create and then ~30-40sec for project to be updated.
I don't understand why we want to waiste 1min if testing we even do not use `--bundle`.

Reply:
We are not creating project for every test. Also we are not testing bundle at all. We have 10 Tns.create in all tests. I don't get why should we refactore all of our tests to use new template. I would like our tests to be executed against real template.

Reply:
10x30sec = 5min :) 
If real is more important than saving 5 min fine.
If we do not sacrifice quality and save 5min -> it will be great and all teams will appreciate it.